strategies/slack.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const passport = require('passport'),
+  SlackStrategy = require('passport-slack').Strategy,
+  { verify, getAuthUrl, getPathOrBase, getCallbackUrl } = require('../utils');
+
+/**
+ * Slack authenticatio strategy
+ *
+ * @param {object} site
+ */
+function createSlackStrategy(site) {
+  passport.use(`slack-${site.slug}`, new SlackStrategy({
+    clientID: process.env.SLACK_CONSUMER_KEY,
+    clientSecret: process.env.SLACK_CONSUMER_SECRET,
+    callbackURL: getCallbackUrl(site, 'slack'),
+    passReqToCallback: true,
+    scope: 'users:read'
+  },
+  verify({
+    username: '_json.user',
+    imageUrl: '_json.info.user.profile.image_1024',
+    name: '_json.info.user.real_name',
+    provider: 'slack'
+  }, site)));
+}
+
+/**
+ * add authorization routes to the router
+ * @param {express.Router} router
+ * @param {object} site
+ * @param {object} provider
+ */
+function addAuthRoutes(router, site, provider) {
+  router.get(`/_auth/${provider}`, passport.authenticate(`${provider}-${site.slug}`));
```

Comment:
I think you should have a helper for this `${provider}-${site.slug}`, you are using it in 9 different places, if the format change one day 😅  i wont be cool